ruby/ext/google/protobuf_c/protobuf.c

```diff
@@ -185,11 +185,16 @@ static void Arena_free(void *data) {
   xfree(arena);
 }
 
+static size_t Arena_memsize(void *data) {
+  Arena *arena = data;
+  return sizeof(Arena) + _upb_ArenaHas(arena->arena);
```

Comment:
I don't know if `_upb_ArenaHas` is the correct function to get the arena memory footprint.

Reply:
It is not, `_upb_ArenaHas()` tells you how much space is currently *available* to allocate on the arena.

`upb_Arena` does not currently have a function for reporting how much memory has been allocated. It would be relatively simple to add one, but there is a complication when the arenas are fused:

```ruby
  # Creating a new message creates an associated arena.
  foo_msg = FooMessage()

  # New message, new arena.
  bar_msg = BarMessage()

  # Joining the two messages "fuses" their arenas.
  # The two arenas now share the same underlying memory.
  # If both arenas reported the full allocated size, we would double-count.
  foo_msg.bar = bar_msg
```

Reply:
Each arena knows its refcount (the number of arenas that have been fused into it).

So perhaps the simplest solution would be to have:

```c
// Returns the total space allocated by the arena, divided
// by the number of live arenas that have been fused into this one.
size_t upb_Arena_SpaceAllocated(const upb_Arena*)
```

Reply:
Hum, I need to dig a bit more on how these Arena work. When you fuse an Arena, into another, it stays alive?